beacon-chain/db/slasherkv/pruning.go

```diff
@@ -11,199 +11,149 @@ import (
 	// there is nothing to prune, so we return early.
-	if lowestEpoch >= endPruneEpoch {
-		log.Debugf("Lowest epoch %d is >= pruning epoch %d, nothing to prune", lowestEpoch, endPruneEpoch)
+	if lowestSlot > endPruneSlot {
+		log.Debugf("Lowest slot %d is > pruning slot %d, nothing to prune", lowestSlot, endPruneSlot)
 		return nil
 	}
 
-	// We prune in increments of `pruningEpochIncrements` at a time to prevent
-	// a long-running bolt transaction which overwhelms CPU and memory.
-	// While we still have epochs to prune based on a cursor, we continue the pruning process.
-	epochAtCursor := lowestEpoch
-	for epochAtCursor < endPruneEpoch {
-		// Each pruning iteration involves a unique bolt transaction. Given pruning can be
-		// a very expensive process which puts pressure on the database, we perform
-		// the process in a batch-based method using a cursor to proceed to the next batch.
-		log.Debugf("Pruned %d/%d epochs worth of attestations", epochAtCursor, endPruneEpoch-1)
-		if err := s.db.Update(func(tx *bolt.Tx) error {
-			rootsBkt := tx.Bucket(attestationDataRootsBucket)
-			attsBkt := tx.Bucket(attestationRecordsBucket)
-			c := rootsBkt.Cursor()
-
-			var lastPrunedEpoch, epochsPruned types.Epoch
-			// We begin a pruning iteration at starting from the first item in the bucket.
-			for k, v := c.First(); k != nil; k, v = c.Next() {
-				// We check the epoch from the current key in the database.
-				// If we have hit an epoch that is greater than the end epoch of the pruning process,
-				// we then completely exit the process as we are done.
-				if !uint64PrefixLessThan(k, encodedEndPruneEpoch) {
-					epochAtCursor = endPruneEpoch
-					return nil
-				}
-
-				epochAtCursor = types.Epoch(binary.LittleEndian.Uint64(k))
-
-				// Attestation in the database look like this:
-				//  (target_epoch ++ _) => encode(attestation)
-				// so it is possible we have a few adjacent objects that have the same slot, such as
-				//  (target_epoch = 3 ++ _) => encode(attestation)
-				// so we only mark an epoch as pruned if the epoch of the current object
-				// under the cursor has changed.
-				if err := rootsBkt.Delete(k); err != nil {
-					return err
-				}
-				if err := attsBkt.Delete(v); err != nil {
-					return err
-				}
-				if epochAtCursor == 0 {
-					epochsPruned = 1
-				} else if epochAtCursor > lastPrunedEpoch {
-					epochsPruned++
-					lastPrunedEpoch = epochAtCursor
-				}
-				slasherAttestationsPrunedTotal.Inc()
-
-				// If we have pruned N epochs in this pruning iteration,
-				// we exit from the bolt transaction.
-				if epochsPruned >= pruningEpochIncrements {
-					return nil
-				}
+	return s.db.Update(func(tx *bolt.Tx) error {
+		// We begin a pruning iteration at starting from the first item in the bucket.
+		proposalBkt := tx.Bucket(proposalRecordsBucket)
+		c := proposalBkt.Cursor()
+		// We begin a pruning iteration at starting from the first item in the bucket.
```

Comment:
```suggestion
		// We begin a pruning iteration starting from the first item in the bucket.
```
On a side note, this comment is the same as on line 124.